aggsender/flows/factory.go

```diff
@@ -43,12 +43,7 @@ func NewFlow(
 		}
 		logger.Infof("Initializing RollupManager contract at address: %s. Genesis block: %d",
 			cfg.RollupManagerAddr, cfg.RollupCreationBlockL1)
-		lerQuerier, err := query.NewLERDataQuerier(
-			cfg.RollupManagerAddr, cfg.RollupCreationBlockL1, rollupDataQuerier)
-		if err != nil {
-			return nil, fmt.Errorf("error creating LER data querier: %w", err)
-		}
-
+		lerQuerier := query.NewLERDataQuerier(cfg.RollupCreationBlockL1, rollupDataQuerier)
```

Comment:
IMO you could extract the instatiation of the common dependencies (e.g. lerQuerier, l2BridgeQuerier, l1InfoTreeQuerier etc.), before the switch-case and reduce duplication (at least slightly).

If you are concerned about instantiating before validating the aggsender mode, we can first validate if the mode is within the valid range of values and if not, return early. WDYT?